internal/jobcontainers/path.go

```diff
@@ -107,7 +104,7 @@ func getApplicationName(commandLine, workingDirectory, pathEnv string) (string,
 		if index == -1 {
 			return "", "", errors.New("no ending quotation mark found in command")
 		}
-		path, err := searchPathForExe(commandLine[1:index+1], searchPath)
+		path, err := searchPathForExe(filepath.Clean(commandLine[1:index+1]), searchPath)
```

Comment:
We can't rely on `filepath.Clean` here as it will simply replace `./ping` -> `ping`, since it assumes the path will be evaluated relative to the working directory, rather than searched for in a list. See [this comment](https://github.com/microsoft/hcsshim/pull/1113/files#r688986162) as well.

Reply:
Probably just need to check for `./` and replace it with `workingDirectory`.

Reply:
For a relative path without a dot (e.g. `ping` or `foo/ping`), we will need to determine the correct behavior. We should understand what the default Windows Server container behavior and CreateProcess behavior is in this case.

Reply:
> We can't rely on `filepath.Clean` here as it will simply replace `./ping` -> `ping`, since it assumes the path will be evaluated relative to the working directory, rather than searched for in a list. See [this comment](https://github.com/microsoft/hcsshim/pull/1113/files#r688986162) as well.

Good point



> Probably just need to check for `./` and replace it with `workingDirectory`.

Check for only a first occurrence or to also handle your mention of "././" or ".\.\" that you made in another comment? I feel like if Windows Server containers don't handle even the first case that might be fine to get away with. "././mybin.exe" seems like an odd command line to provide anyways. "../mybin.exe", "../../mybin.exe" however seems like it actually has its uses, but now we're back to your comment of writing custom parsing code

Reply:
> For a relative path without a dot (e.g. `ping` or `foo/ping`), we will need to determine the correct behavior. We should understand what the default Windows Server container behavior and CreateProcess behavior is in this case.

I think we should be pretty inline with the Windows Server container behavior. It seemed like everything was behaving the same before the relative path with dot discussion was brought up and the `filepath.Clean` change.. afaik cexecsvc tries to mirror the `CreateProcess` behavior and we tried to mirror the cexecsvc behavior, so they should be essentially the same before this work in regards to how they handle relative paths with no dot. Granted I'd have to go check `CreateProcess`'s logic to see just how similar they actually are.

Reply:
I think this all comes back to whether we want to support relative paths or not. The answer to that question will inform our approach here. Does the host process KEP say anything about relative paths?

Reply:
Not that I could find. There's a mention that to access container mounts you could either use a relative path, but nothing regarding the cmdline. https://github.com/kubernetes/enhancements/blob/master/keps/sig-windows/1981-windows-privileged-container-support/README.md#container-mounts

Reply:
Ok after syncing with James, we didn't call out anywhere that relative paths with a dot would work for these, and they don't work already for ws contains, so I'm just going to omit this. I'll change this PR to updating the tests, and a small fix for the environment variable replacement logic